mne/decoding/base.py

```diff
@@ -709,14 +709,12 @@ def _set_cv(cv, estimator=None, X=None, y=None):
 def _check_estimator(estimator, get_params=True):
     """Check whether an object has the fit, transform, fit_transform and
     get_params methods required by scikit-learn"""
-    err = True
-    for attr in ('predict', 'transform', 'predict_proba', 'decision_function'):
-        if hasattr(estimator, attr):
-            err = False
-    if not hasattr(estimator, 'fit'):
-        err = True
-
-    if err:
+    valid_methods = ('predict', 'transform', 'predict_proba',
+                     'decision_function')
+    if (
+        (not hasattr(estimator, 'fit')) or
+        (not np.any([hasattr(estimator, method) for method in valid_methods]))
```

Comment:
`np.any` is proably overkill here compared to my original suggestion, I think it will make an array whereas `any(hasattr(...) ...)` operates as a generator and short-circuits as soon as it finds one (and is also fewer chars, and uses a builtin, etc.)